File: tests/filetypes/sample.cs
// C# example to test editor settings

using System;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        // Variables and constants
        string name = "C#";
        const double version = 10.0;

        // Print a message
        Console.WriteLine($"Hello, {name}! Version: {version}");

        // Call a method
        int result = Add(5, 10);
        Console.WriteLine($"Sum of 5 and 10: {result}");

        // Conditional statement
        if (result > 10)
        {
            Console.WriteLine("Result is greater than 10");
        }
        else
        {
            Console.WriteLine("Result is less than or equal to 10");
        }

        // Arrays and loops
        int[] numbers = { 1, 2, 3, 4, 5 };
        foreach (int num in numbers)
        {
            Console.WriteLine($"Number: {num}");
        }

        // Dictionary (similar to map)
        Dictionary<string, object> person = new Dictionary<string, object>
        {
            { "name", "Alice" },
            { "age", 30 },
            { "email", "alice@example.com" }
        };
        Console.WriteLine($"{person["name"]} is {person["age"]} years old.");

        // Call a method from a class
        Person person1 = new Person("Bob", 35);
        Console.WriteLine(person1.Greet());

        // Handling null values with nullable types
        string? optionalString = "This is a nullable string";
        if (optionalString != null)
        {
            Console.WriteLine($"Optional string: {optionalString}");
        }

        // Lambda expression
        Func<int, int> square = x => x * x;
        Console.WriteLine($"Square of 4: {square(4)}");

        // Try-catch for exception handling
        try
        {
            FetchData("");
        }
        catch (Exception e)
        {
            Console.WriteLine($"Error fetching data: {e.Message}");
        }
    }

    // Method to add two numbers
    static int Add(int a, int b)
    {
        return a + b;
    }

    // Method that simulates fetching data and throws an exception
    static void FetchData(string url)
    {
        if (string.IsNullOrEmpty(url))
        {
            throw new ArgumentException("URL cannot be empty");
        }
        Console.WriteLine($"Fetching data from {url}");
    }
}

// Class with a method
class Person
{
    public string Name { get; set; }
    public int Age { get; set; }

    public Person(string name, int age)
    {
        Name = name;
        Age = age;
    }

    public string Greet()
    {
        return $"Hello, my name is {Name} and I am {Age} years old.";
    }
}

File: tests/filetypes/sample.pl
# Perl example to test editor settings

# Variables and constants
my $name = "Perl";
my $version = 5.32;

# Print a message
print "Hello, $name! Version: $version\n";

# Function to add two numbers
sub add {
    my ($a, $b) = @_;
    return $a + $b;
}

# Call the function
my $result = add(5, 10);
print "Sum of 5 and 10: $result\n";

# Conditional statement
if ($result > 10) {
    print "Result is greater than 10\n";
} else {
    print "Result is less than or equal to 10\n";
}

# Arrays and loops
my @numbers = (1, 2, 3, 4, 5);
foreach my $num (@numbers) {
    print "Number: $num\n";
}

# Hash (similar to a dictionary)
my %person = (
    name  => "Alice",
    age   => 30,
    email => "alice\@example.com"
);
print "$person{name} is $person{age} years old.\n";

# Using subroutines
sub greet {
    my ($name, $age) = @_;
    return "Hello, my name is $name and I am $age years old.";
}

my $greeting = greet("Bob", 35);
print "$greeting\n";

# Error handling using eval
eval {
    fetch_data("");
};
if ($@) {
    print "Error fetching data: $@\n";
}

# Function that throws an error
sub fetch_data {
    my ($url) = @_;
    die "URL cannot be empty" if $url eq '';
    print "Fetching data from $url\n";
}

